docs/NETWORK-ACLS.md

```diff
@@ -0,0 +1,289 @@
+# Network ACLs
+
+An access control list (ACL) consists of one or more access control entries (ACEs) that collectively define
+the network traffic profile.
+This document describes different types of ACEs, their semantics, limitations and provides few basic examples
+for common use-cases.
+
+## Configuration Model
+
+In EVE, ACLs are configured separately for every ECO interface.
+Protobuf-modeled configuration model of ACLs can be found in the [fw.proto file](../api/proto/config/fw.proto).
+To control the network traffic between a given ECO and other endpoints within the scope of a given network instance,
+add one or more ACEs under the `acls` field of [NetworkAdapter](../api/proto/config/netconfig.proto).
+
+Every ACE should have a unique non-zero integer ID assigned. Please note that ACE ID should not exceed 24bits.
+ACE ID is set in [flowlogs](../api/proto/flowlog/flowlog.proto) as `aclId` for every flow to show which ACE was applied.
+ACE `ID=0` is reserved for a default reject-all rule that implicitly exists at the end of every ACL (even flows whose
+packets are rejected are logged).
+
+## ACL Semantics
+
+Every ECO interface is assigned a single separate access control list filtering inbound and outbound traffic
+(unless it is a passthrough interface, in which case EVE networking is completely bypassed).
+A single entry of ACL, denoted as ACE, consists of a set of matches that use layer 3 and 4 information to select
+a subset of the traffic, and a set of actions deciding what to do with the selected traffic. Optionally, it is possible
+to limit ACE to only ingress or egress flow direction. By default, ACEs are bi-directional, i.e. applied irrespective
+of the flow direction.
+
+For every packet leaving or entering ECO, the list of ACEs assigned to the corresponding interface is traversed
+sequentially in the order as configured. The fate of a packet is determined by the first ACE in the list for which
+the L3 & L4 packet header fields satisfy every match entry (i.e., matches are ANDed and empty set matches any packet).
+Packet can be allowed to continue (`ALLOW` action), dropped (`DROP` action) or mangled - specifically
+the destination IP address and port can be rewritten to allow an inbound flow to traverse the NAT of a local network
+(`PORTMAP` action).
```

Comment:
Would make sense to state that in the API the DROP action is the default (applies if the action is empty).

Reply:
Default ACE action is actually ALLOW:  https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/zedrouter/acl.go#L910
For example, this is how eden configured allow-all ACL: https://github.com/lf-edge/eden/blob/master/pkg/expect/networkInstance.go#L182-L197

Reply:
Oh, you're correct. I wanted to type less back when I generated json files by hand and with the implicit drop at the end that's what I did. Anyhow, makes sense to state the default when there is no action specified.

Reply:
It is mentioned 3 paragraphs below:
"ACE which is configured without specifying the action.."